### PR TITLE
refactor(move queries to findhandler)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

In preparation for the refaction of the latest_analyses function some code needed to be moved from other apis to the fund handler.

### Added

- functions `get_latest_nipt_analysis_to_upload` and `get_latest_microsalt_analysis_to_upload`
- tests for these functions

### Changed

- Moved code from nipt.py and microsalt/base.py to find_business_data.py. The code that got moved was placed into functions which are now at those initial places. 

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store/api/find/test_find_business_data_analysis.py [hunk 1]
```diff
@@ -23,3 +24,39 @@ def test_get_analysis_by_case(
     for analysis in analyses:
         assert analysis
         assert analysis.family == case
+
+
+def test_get_latest_nipt_analysis_to_upload(
```
Comment (diitaz93) [inline tests/store/api/find/test_find_business_data_analysis.py:29]: and add docstrings to these methods too

**Review by diitaz93 — APPROVED** [review]
looks good 👍

File: cg/store/api/find_business_data.py [hunk 2]
```diff
@@ -94,6 +94,19 @@ def analyses_ready_for_vogue_upload(
 
         return records
 
+    def get_latest_nipt_analysis_to_upload(self):
```
Comment (seallard) [inline cg/store/api/find_business_data.py:97]: type hints

**Review by seallard — APPROVED** [review]
👍

File: cg/store/api/find_business_data.py (hunk 2 above)
Comment (ChrOertlin) [reply cg/store/api/find_business_data.py:97]: These functions will be refactored immediately in my other pr.

**Pull request merged**